src/Modules/Backend/Domain/Action/AbstractActionController.php

```diff
@@ -66,11 +77,63 @@ final public static function getActionSlug(): ActionSlug
 
     public function __invoke(Request $request): Response
     {
+        if ($this->needsRedirectToProfilePage($request)) {
+            return $this->redirectToProfilePage($request);
+        }
+
         $this->execute($request);
 
         return $this->getResponse($request);
     }
 
+    private function redirectToProfilePage(Request $request): Response
+    {
+        /** @var User $user */
+        $user = $this->tokenStorage->getToken()->getUser();
+
+        $this->header->addFlashMessage(
+            new FlashMessage(
+                'msg.2FAIsRequired',
+                FlashMessageType::WARNING
+            )
+        );
+
+        return new RedirectResponse(
+            UserEdit::getActionSlug()->generateRoute($this->router) . '/' . $user->getId()
+        );
+    }
+
+    private function needsRedirectToProfilePage(Request $request): bool
```

Comment:
`shouldUserConfigure2FA` would be a more descriptive method name in my opinion. Now it can be any reason on why the redirect need to happen.

Reply:
fixed bf2f1736aee2591f42f0167ece3c12c3d26c248f